### ISSUE TITLE
Regression on tests (native builds, not asmjs or webassembly)

### ISSUE THREAD

**Issue by philipp-classen** [cliqz-oss/anonymous-credentials#14]
Originally reported in https://github.com/cliqz-oss/anonymous-credentials/pull/13

Splitting it, as the failing tests are independent of the PR that adds the CI.

**Comment by acatarineu** [comment]
>From what I can see, when passing the data from JavaScript to the native library, it adds a the number of arguments + 1 as the last parameter. That it is why it complains that the data is not an Uint8Array.
Hrm, where do you see that? I could not see the number of args + 1 as last parameter.

If that's not the issue, I assume that it should be relatively easy to find out why `GS_GET_DATA` is returning `NULL` in https://github.com/cliqz-oss/anonymous-credentials/blob/master/groupsign_napi.c#L177. Maybe something changed in newer nodejs versions? Or maybe a special case of `new Uint8Array()` being passed?

**Comment by y3ti** [comment]
@acatarineu we added a simple CI workflow in GitHub Actions. You can see the error here:
https://github.com/cliqz-oss/anonymous-credentials/runs/2197688640

**Comment by acatarineu** [comment]
Yes sorry, I was referring to the `it adds a the number of arguments + 1` observation from https://github.com/cliqz-oss/anonymous-credentials/pull/13#issuecomment-807655828, I could not find it in the code.

**Comment by philipp-classen** [comment]
@acatarineu I hope to find time later in the day to debug into it. What I could only see so far by adding console.log statements is that the argument list ends with a number and that the check for Uint8Array fail. But it also only happens in the native build, not the other builds.

I hope to check back with more information, but so far I only know that these here throws:

https://github.com/cliqz-oss/anonymous-credentials/blob/9d76ca7f34a4658661c02af827548d02b2911247/pre.js#L79

And the calling code is that here:
https://github.com/cliqz-oss/anonymous-credentials/blob/9d76ca7f34a4658661c02af827548d02b2911247/tests/tests.js#L172

Note that we are explicitly passing a Uint8Array, so I'm confused that the native code thinks not all arguments are Uint8Array. And I'm even more confused that the happy path is not affected.

**Comment by acatarineu** [comment]
But only nodejs native fails? Then it's not `pre.js`, but `groupsign_napi.c`, right? Or I'm missing something?

**Comment by acatarineu** [comment]
If I recall correctly, `pre.js` are bindings only for wasm/asmjs.

**Comment by philipp-classen** [comment]
Note: Production is not affected (I don't see any problems after releasing from master)

When debugging in, some observations:
* signer.seed(new Uint8Array()) --> input data must be uint8array
* signer.seed(new Uint8Array(0)) --> input data must be uint8array
* signer.seed(new Uint8Array([])) --> input data must be uint8array
* signer.seed(new Uint8Array(1)) --> seed too small
* signer.seed(new Uint8Array([0])) --> seed too small

For some reason, empty Uint8Array trigger the error, but only in native builds.

**Comment by philipp-classen** [comment]
Will later look into it more. I now also realized that I'm using Node 14.16.0 to run the tests. The same version that we use in production.

With Node 12, it passes. So, it is a regression with Node 14. Fortunately, we are not hit by it in production, but at least that should help us to track it down.

**Comment by philipp-classen** [comment]
@acatarineu Will later check (it is not super urgent to fix it now), but that could be the reason (v14 changes):

```
New V8 ArrayBuffer API

    src: migrate to new V8 ArrayBuffer API (Thang Tran) #30782

Multiple ArrayBuffers pointing to the same base address are no longer allowed by V8. This may impact native addons.
```
